source/Octopus.Client/IOctopusAsyncRepository.cs

```diff
@@ -1,68 +1,82 @@
         IMachineRoleRepository MachineRoles { get; }
-        IMigrationRepository Migrations { get; }
-        IOctopusServerNodeRepository OctopusServerNodes { get; }
-        IPerformanceConfigurationRepository PerformanceConfiguration { get; }
         IProjectGroupRepository ProjectGroups { get; }
         IProjectRepository Projects { get; }
         IProjectTriggerRepository ProjectTriggers { get; }
         IProxyRepository Proxies { get; }
         IReleaseRepository Releases { get; }
         IRetentionPolicyRepository RetentionPolicies { get; }
-        ISchedulerRepository Schedulers { get; }
-        IServerStatusRepository ServerStatus { get; }
-        ISpaceRepository Spaces { get; }
         ISubscriptionRepository Subscriptions { get; }
         ITagSetRepository TagSets { get; }
-        ITaskRepository Tasks { get; }
-        ITeamsRepository Teams { get; }
         ITenantRepository Tenants { get; }
         ITenantVariablesRepository TenantVariables { get; }
-        IUserRepository Users { get; }
-        IUserRolesRepository UserRoles { get; }
         IVariableSetRepository VariableSets { get; }
         IWorkerPoolRepository WorkerPools { get; }
         IWorkerRepository Workers { get; }
+        SpaceRootResource SpaceRootDocument { get; }
+    }
+
+    public interface ISystemScopedRepository
+    {
+        ISchedulerRepository Schedulers { get; }
+        IServerStatusRepository ServerStatus { get; }
+        ISpaceRepository Spaces { get; }
+        IUserRepository Users { get; }
+        IUserRolesRepository UserRoles { get; }
+        IBackupRepository Backups { get; }
+        IMigrationRepository Migrations { get; }
+        IOctopusServerNodeRepository OctopusServerNodes { get; }
+        IPerformanceConfigurationRepository PerformanceConfiguration { get; }
+        IConfigurationRepository Configuration { get; }
+        ICommunityActionTemplateRepository CommunityActionTemplates { get; }
+        IFeaturesConfigurationRepository FeaturesConfiguration { get; }
+    }
+
+    /// <summary>
+    /// A simplified interface to commonly-used parts of the API.
+    /// Functionality not exposed by this interface can be accessed
+    /// using <see cref="IOctopusAsyncRepository.Client" />.
+    /// </summary>
+    public interface IOctopusAsyncRepository : ISpaceScopedRepository, ISystemScopedRepository
+    {
+        RootResource RootDocument { get; }
+        IEventRepository Events { get; }
+        ITaskRepository Tasks { get; }
+        ITeamsRepository Teams { get; }
         IScopedUserRoleRepository ScopedUserRoles { get; }
         IUserPermissionsRepository UserPermissions { get; }
+        IOctopusAsyncClient Client { get; }
+        SpaceContext SpaceContext { get; }
+        Task<ISystemScopedRepository> ForSystemContext();
+        Task<ISpaceScopedRepository> ForSpaceContext(string spaceId);
+        Task<IOctopusAsyncRepository> ForSpaceAndSystemContext(string spaceId);
```

Comment:
Users can do context switch from the root, they can only access the context specific repositories once they switched